compiler/src/dotty/tools/dotc/parsing/Scanners.scala

```diff
@@ -385,12 +388,11 @@ object Scanners {
     /** Produce next token, filling TokenData fields of Scanner.
      */
     def nextToken(): Unit =
-      val lastToken = token
```

Comment:
This will slow down nextToken, which is a very hot method, and might in fact be the most crucial method for performance in some usage scenarios (like when scanning a large body of code). I think fixing rewrite logic should not make Scanner slower by whatever margin. 

Can we identify why we need a different Scanning functionality for rewrites, and what is the minimal change needed?

Why does Scanner have to change at all?